SCAutolib/controller.py

```diff
@@ -82,9 +102,20 @@ def prepare(self, force: bool, gdm: bool, install_missing: bool,
             self.setup_ipa_client(force=force)
         except exceptions.SCAutolibWrongConfig as e:
             logger.info(e)
+
         for usr in self.lib_conf["users"]:
-            u = self.setup_user(usr, force=force)
-            self.enroll_card(u)
+            self.setup_user(usr, force=force)
+
+        # Create cards defined in config. For physical cards only objects will
+        # be created while for virtual cards tokens will be created and enrolled
+        for token in self.lib_conf["cards"]:
+            # prepare CA objects for physical cards
+            if token["card_type"] == CardType.physical:
+                self.setup_custom_ca(token)
+                self.setup_card(token)
+            elif token["card_type"] == CardType.virtual:
+                c = self.setup_card(token)
+                self.enroll_card(c)
```

Comment:
To reduce the area for unexpected errors it would be nice to add also `else` statement that will rise an  exception on unknown card type